OpenMarket/OpenMarket/Networking/API/APIProtocol.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  APIProtocol.swift
+//  OpenMarket
+//
+//  Created by 전민수 on 2022/07/26.
+//
+
+import Foundation
+
+protocol APIProtocol {
+    var configuration: APIConfiguration { get }
+}
+
+extension APIProtocol {
+    func retrieveProduct<T: Decodable>(using client: APIClient = APIClient.shared,
+                                    dataType: T.Type,
+                                    completion: @escaping (Result<T,APIError>) -> Void) {
+        
+        var request = URLRequest(url: configuration.url)
+        request.httpMethod = configuration.method.rawValue
```

Comment:
```suggestion
        request.httpMethod = HttpMethod.get.rawValue
```

retrieve 면 method 가 get 이라 이렇게 명시적으로 할당 가능할텐데 configuration의 method 값을 할당하고 있네요!
configuration에 method 프로퍼티가 필요없는 건 아닌지 생각이 듭니다~ 🧐